wagtail/admin/static_src/wagtailadmin/js/core.js

```diff
@@ -176,6 +180,7 @@ $(function() {
     $(document).on('click', '.tab-nav a', function(e) {
         e.preventDefault();
         $(this).tab('show');
+        history.pushState(null, null, $(this).attr('href'));
```

Comment:
This should either use `history.replaceState`, or also add a `popstate` event handler to the window so navigating back and forth with the browser back button works as expected.